pkg/instrumentation/podmutator.go

```diff
@@ -329,36 +298,27 @@ func (pm *instPodMutator) Mutate(ctx context.Context, ns corev1.Namespace, pod c
 		return pod, nil
 	}
 
+	generalContainerNames := annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectContainerName)
+	insts.setInstrumentationLanguageContainers(generalContainerNames)
+
 	// We retrieve the annotation for podname
 	if pm.config.EnableMultiInstrumentation() {
 		// We use annotations specific for instrumentation language
-		insts.Java.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectJavaContainersName)
-		insts.NodeJS.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectNodeJSContainersName)
-		insts.Python.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectPythonContainersName)
-		insts.DotNet.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectDotnetContainersName)
-		insts.Go.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectGoContainersName)
-		insts.ApacheHttpd.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectApacheHttpdContainersName)
-		insts.Nginx.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectNginxContainersName)
-		insts.Sdk.Containers = annotationValue(ns.ObjectMeta, pod.ObjectMeta, annotationInjectSdkContainersName)
+		insts.Java.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectJavaContainersName)
+		insts.NodeJS.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectNodeJSContainersName)
+		insts.Python.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectPythonContainersName)
+		insts.DotNet.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectDotnetContainersName)
+		insts.Go.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectGoContainersName)
+		insts.ApacheHttpd.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectApacheHttpdContainersName)
+		insts.Nginx.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectNginxContainersName)
+		insts.Sdk.Containers = appendContainers(generalContainerNames, ns.ObjectMeta, pod.ObjectMeta, annotationInjectSdkContainersName)
```

Comment:
I would feel a bit better if we operated on slices of strings here instead of a single string containing a comma separated list of values. Do you think we could do that without making the code more complicated?

Reply:
Yes. I don't like the current "string list" too much. Let me check if I can do something to improve it...

Reply:
@swiatekm I added some changes to improve how this is managed (and avoid some `splits`, `joins` in multiple parts of the code). I think it looks better now.